### PR TITLE
S130 update gcp example

### PR DESCRIPTION
**Pull request by eschultink**
### Fixes
  - included bulk and asana/slack/zoom cases in GCP examples; fix and align features was AWS
  - Google Meet spec (lost??)
  - some secret's referenced by full resource ID, in contexts where GCP tf modules need just the short id
 

### Change implications

 - dependencies added/changed? **no**

### CODE REVIEW

File: infra/examples-dev/gcp-google-workspace/variables.tf [hunk 1]
```diff
@@ -33,15 +40,26 @@ variable "connector_display_name_suffix" {
...
 
 variable "google_workspace_example_user" {
   type        = string
   description = "user to impersonate for Google Workspace API calls (null for none)"
 }
+
+variable "gcp_region" {
+  type        = string
+  description = "region in which to provision GCP resources, if applicable"
+  default     = "us-central1"
```
Comment (jlorper) [inline infra/examples-dev/gcp-google-workspace/variables.tf:64]: maybe this shouldn't have a default to force the user to choose one?

File: infra/examples/gcp-google-workspace/main.tf [hunk 2]
```diff
@@ -10,12 +10,39 @@ terraform {
...
 
+locals {
+  base_config_path = "${var.psoxy_base_dir}/configs/"
+  bulk_sources = {
+    "hris" = {
+      source_kind = "hris"
+      rules = {
+        columnsToRedact = []
+        columnsToPseudonymize = [
+          "email",
```
Comment (jlorper) [inline infra/examples/gcp-google-workspace/main.tf:21]:
```suggestion
          "employee_email",
```

File: infra/examples/gcp-google-workspace/main.tf [hunk 3]
```diff
@@ -10,12 +10,39 @@ terraform {
...
+          "employee_id"
+        ]
+      }
+    },
+    "qualtrics" = {
+      source_kind = "qualtrics"
+      rules = {
+        columnsToRedact = []
+        columnsToPseudonymize = [
+          "email",
```
Comment (jlorper) [inline infra/examples/gcp-google-workspace/main.tf:31]: This qualtrics file probably doesn't matter that much, but ideally should be :point_down: to follow our conventions
```suggestion
          "employee_email",
```

File: infra/examples/gcp-google-workspace/README.md [hunk 4]
```diff
@@ -51,3 +51,8 @@ terraform apply
 
 Review the plan and confirm to apply.
 
+## Notes
+
+Deployment will create:
+  - one GCS bucket for staging artifacts
```
Comment (aperez-worklytics) [inline infra/examples/gcp-google-workspace/README.md:57]: Just a suggestion

```suggestion
  - one GCS bucket for staging artifacts with the binaries to deploy
```

File: infra/examples/gcp-google-workspace/main.tf [hunk 5]
```diff
@@ -41,12 +68,14 @@ resource "google_project" "psoxy-project" {
 
 module "psoxy-gcp" {
   # source = "../../modules/gcp"
-  source = "git::https://github.com/worklytics/psoxy//infra/modules/gcp?ref=v0.4.2"
+  source = "git::https://github.com/worklytics/psoxy//infra/modules/gcp?ref=v0.4.3"
```
Comment (jlorper) [inline infra/examples/gcp-google-workspace/main.tf:71]: source 0.4.3 everywhere, but `psoxy_version     = "0.4.2"`. Is that right?

File: infra/examples-dev/gcp-google-workspace/main.tf [hunk 6]
```diff
@@ -31,35 +60,33 @@ module "worklytics_connector_specs" {
...
+  # org_id          = var.gcp_org_id # if project is in a GCP folder, this value is implicit and this line should be commented out
 }
 
 module "psoxy-gcp" {
   source = "../../modules/gcp"
 
   project_id        = google_project.psoxy-project.project_id
   invoker_sa_emails = var.worklytics_sa_emails
+  psoxy_base_dir    = var.psoxy_base_dir
+  psoxy_version     = "0.4.2"
```
Comment (jlorper) [inline infra/examples-dev/gcp-google-workspace/main.tf:75]: 0.4.3 ??

File: infra/examples/gcp-google-workspace/main.tf [hunk 7]
```diff
@@ -115,8 +144,83 @@ module "worklytics-psoxy-connection" {
...
+  display_name       = "${title(each.key)}${var.connector_display_name_suffix} via Psoxy"
+}
+
+# BEGIN LONG ACCESS AUTH CONNECTORS
+
+resource "google_service_account" "long_auth_connector_sa" {
+  for_each = module.worklytics_connector_specs.enabled_oauth_long_access_connectors
+
+  project      = google_project.psoxy-project.project_id
+  account_id   = "psoxy-${each.key}"
```
Comment (jlorper) [inline infra/examples/gcp-google-workspace/main.tf:159]: seen this limit in the example. Are SA names limited to 30 chars?
```suggestion
  account_id   = "psoxy-${substr(each.key, 0, 24)}"
```

File: infra/modules/gcp-psoxy-bulk/main.tf [hunk 8]
```diff
@@ -37,22 +49,10 @@ resource "google_storage_bucket" "output-bucket" {
...
-      labels
-    ]
-  }
-}
-
-resource "google_secret_manager_secret_iam_member" "salt-secret-access-for-service-account" {
-  member    = "serviceAccount:${google_service_account.service-account.email}"
-  role      = "roles/secretmanager.secretAccessor"
-  secret_id = var.salt_secret_id
+  account_id   = local.function_name
```
Comment (jlorper) [inline infra/modules/gcp-psoxy-bulk/main.tf:53]: again, should we limit to 30 chars?

File: infra/modules/gcp-psoxy-bulk/main.tf [hunk 9]
```diff
@@ -69,30 +69,34 @@ resource "google_storage_bucket_iam_member" "grant_sa_read_on_processed_bucket"
...
+resource "google_secret_manager_secret_iam_member" "grant_sa_accessor_on_secret" {
+  for_each = local.secret_bindings
+
+  project   = var.project_id
+  secret_id = each.value.secret_id
+  member    = "serviceAccount:${google_service_account.service-account.email}"
+  role      = "roles/secretmanager.secretAccessor"
+}
+
 resource "google_cloudfunctions_function" "function" {
```
Comment (jlorper) [inline infra/modules/gcp-psoxy-bulk/main.tf:97]: doubt: are we going to have the same issue with the tags? maybe we should add `lifecycle ignore_changes` everywhere too

File: infra/modules/gcp-sa-auth-key-aws-secret/main.tf [hunk 10]
```diff
@@ -27,4 +27,10 @@ resource "aws_ssm_parameter" "value" {
   type        = "SecureString"
   description = "Key for gcp service account ${var.service_account_id}"
   value       = google_service_account_key.key.private_key
+
+  lifecycle {
+    ignore_changes = [
+      tags,
```
Comment (jlorper) [inline infra/modules/gcp-sa-auth-key-aws-secret/main.tf:33]: :+1: just saw this

**Review by aperez-worklytics — APPROVED** [review]

**Review by jlorper — APPROVED** [review]
Few suggestions, none of them a real blocker.

File: infra/modules/gcp-sa-auth-key-aws-secret/main.tf (hunk 10 above)
Comment (eschultink) [reply infra/modules/gcp-sa-auth-key-aws-secret/main.tf:33]: yeah - the `tags` isn't such a big deal.  It just seems that many customers use `tags` in practice, so after they run our terraform stuff they go through and tag stuff (sometimes manually, or with asset discovery scripts - I'm not really clear) - but either way I don't want to overwrite their changes - nor to I want to add the extensibility to our scripts to pass through all possible tags they might want to encode.

**Pull request merged**